File: app/src/main/java/com/kyhsgeekcode/disassembler/HexView.java
package com.kyhsgeekcode.disassembler;

import android.content.Context;
import android.graphics.Canvas;
import android.view.MotionEvent;
import android.view.View;

public class HexView extends View {
    boolean isScrolling = false;
    private byte[] bytes;

    public HexView(Context context) {
        super(context);
    }

    public HexView(Context context, android.util.AttributeSet attrs) {
        super(context, attrs);
    }

    public HexView(Context context, android.util.AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public HexView(Context context, android.util.AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    public void draw(Canvas canvas) {
        super.draw(canvas);
        int w = getMeasuredWidth();
        int h = getMeasuredHeight();
        //if min w->scrollable
        //if min h->scroll
        return;
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN:
                isScrolling = true;
                break;
            case MotionEvent.ACTION_MOVE:
                if (isScrolling) {

                }
                break;
            case MotionEvent.ACTION_UP:
                isScrolling = false;
                break;
        }
        return true;
    }


    public void SetBytes(byte[] bytes) {
        this.bytes = bytes;
    }

}

File: app/src/main/java/com/kyhsgeekcode/disassembler/HexView.kt
package com.kyhsgeekcode.disassembler

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.View

class HexView : View {
    var isScrolling = false
    private lateinit var bytes: ByteArray

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
    }

    //    public HexView(Context context, android.util.AttributeSet attrs, int defStyleAttr, int defStyleRes) {
    ////        super(context, attrs, defStyleAttr, defStyleRes);
    //    }
    override fun draw(canvas: Canvas) {
        super.draw(canvas)
        val w = measuredWidth
        val h = measuredHeight
        //if min w->scrollable
        //if min h->scroll
        return
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        when (event.action) {
            MotionEvent.ACTION_DOWN -> isScrolling = true
            MotionEvent.ACTION_MOVE -> if (isScrolling) {
            }
            MotionEvent.ACTION_UP -> isScrolling = false
        }
        return true
    }

    fun SetBytes(bytes: ByteArray) {
        this.bytes = bytes
    }
}